src/Xamarin.Android.Build.Tasks/Tasks/Aapt.cs

```diff
@@ -115,34 +141,47 @@ public override bool Execute ()
 			if (CreatePackagePerAbi)
 				Log.LogDebugMessage ("  SupportedAbis: {0}", SupportedAbis);
 
-			bool upToDate = ManifestIsUpToDate (ManifestFile);
-
 			if (ResourceNameCaseMap != null)
 				foreach (var arr in ResourceNameCaseMap.Split (';').Select (l => l.Split ('|')).Where (a => a.Length == 2))
 					resource_name_case_map [arr [1]] = arr [0]; // lowercase -> original
 
-			if (AdditionalAndroidResourcePaths != null)
-				foreach (var dir in AdditionalAndroidResourcePaths)
-					if (!string.IsNullOrEmpty (dir.ItemSpec))
-						upToDate = upToDate && ManifestIsUpToDate (string.Format ("{0}{1}{2}", dir, Path.DirectorySeparatorChar, "manifest", Path.DirectorySeparatorChar, "AndroidManifest.xml"));
+			ThreadingTasks.Parallel.ForEach (ManifestFiles, () => 0, (manifestFile, loopState, localSum) => {
 
-			if (upToDate) {
-				Log.LogMessage (MessageImportance.Normal, "  Additional Android Resources manifsets files are unchanged. Skipping.");
-				return true;
-			}
+				if (!File.Exists (manifestFile.ItemSpec)) {
+					LogDebugMessage ("{0} does not exists. Skipping", manifestFile.ItemSpec);
+					return 0;
+				}
 
-			ExecuteForAbi (null);
+				bool upToDate = ManifestIsUpToDate (manifestFile.ItemSpec);
+
+				if (AdditionalAndroidResourcePaths != null)
+					foreach (var dir in AdditionalAndroidResourcePaths)
+						if (!string.IsNullOrEmpty (dir.ItemSpec))
+							upToDate = upToDate && ManifestIsUpToDate (string.Format ("{0}{1}{2}", dir, Path.DirectorySeparatorChar, "manifest", Path.DirectorySeparatorChar, "AndroidManifest.xml"));
+
+				if (upToDate) {
+					LogMessage ("  Additional Android Resources manifsets files are unchanged. Skipping.");
+					return 0;
+				}
+
+				ExecuteForAbi (GenerateCommandLineCommands (manifestFile.ItemSpec, null));
```

Comment:
We could "merge" this and the following `if`:

    var sabis = SupportedAbis.Split (new char [] { ',', ';' }, StringSplitOptions.RemoveEmptyEntries);
    var def   = new string[]{null};
    var abis  = (CreatePackagePerAbi && sabis.Length > 1) ? def.Concat (abis) : def;
    foreach (var abi in abis) {
        ExecuteForAbi (GenerateCommandLineCommands (manifestFile.ItemSpec, abi));
    }

This would reduce special-casing and help ensure that if we fix for one we fix for all.